src/containers/forms/OrganizationForm/index.tsx

```diff
@@ -0,0 +1,191 @@
+  redirectAfterAction: string;
+}
+
+export const defaultInitialValues: OrganizationFormFields = {
+  active: false,
+  identifier: '',
+  name: '',
+  type: defaultOrganizationType,
+};
+
+// TODO - indicator that a team has been added to store??
+/** Organization form component */
+const OrganizationForm = (props: OrganizationFormProps) => {
+  /** track when redirection from this form page should occur */
+  const [ifDoneHere, setIfDoneHere] = useState<boolean>(false);
+  const { initialValues, redirectAfterAction, disabledFields } = props;
+  const [globalError, setGlobalError] = useState<string>('');
+
+  /** edit mode set to true if we are updating a team data. */
+  const editMode: boolean = initialValues.identifier !== '';
+
+  return (
+    <div className="form-container">
+      {ifDoneHere && <Redirect to={redirectAfterAction} />}
+      <Formik
+        initialValues={initialValues}
+        validationSchema={OrgSchema}
+        // tslint:disable-next-line: jsx-no-lambda
+        onSubmit={(values, { setSubmitting, setFieldValue }) => {
+          const organizationService = new OpenSRPService(OPENSRP_ORGANIZATION_ENDPOINT);
+          setFieldValue(
+            'identifier',
+            generateNameSpacedUUID(`${moment().toString()}`, OrgFormNameSpace)
+          );
+
+          if (editMode) {
+            // 2 calls for each for updating team information and updating practitioner_role table
+            organizationService
+              .update(values)
+              .then(() => {
+                setSubmitting(false);
+                setIfDoneHere(true);
+              })
+              .catch((e: Error) => {
+                setGlobalError(e.message);
+                setSubmitting(false);
+              });
+          } else {
+            organizationService
+              .create(values)
+              .then(() => {
+                setSubmitting(false);
+                setIfDoneHere(true);
+              })
+              .catch((e: Error) => {
+                setGlobalError(e.message);
+                setSubmitting(false);
+              });
+          }
+        }}
+      >
+        {({ errors, isSubmitting, setFieldValue, values }) => (
+          <Form className="mb-5">
+            <FormGroup className="non-field-errors">
+              {globalError !== '' && <p className="form-text text-danger">{globalError}</p>}
+            </FormGroup>
+            <FormGroup>
+              <Label>Name</Label>
+              <Field
+                type="text"
+                name="name"
+                id="name"
+                disabled={disabledFields.includes('name')}
+                className={errors.name ? `form-control is-invalid` : `form-control`}
+              />
+              <ErrorMessage
+                name="name"
+                component="small"
+                className="form-text text-danger name-error"
+              />
+            </FormGroup>
+            <FormGroup>
```

Comment:
Is there a reason why you are using radio buttons instead of one checkbox?

Reply:
Seemed like a good idea at the time

Reply:
Could / should there be other status we would want to demarcate here? (eg `Pending`, `Closed`, etc?)

If not then let's make this a checkbox.

Reply:
I don't think this should be a blocker to merging.

Reply:
@p-netm @cKellyDesign can we create an issue to at least discuss this?  If we can use a checkbox idk what valid reasons we would have not to.